zppy/templates/pcmdi_diags.bash

```diff
@@ -0,0 +1,1612 @@
+    proc = Popen(p, stdout=PIPE, shell=True)
+    if multiprocessing == True:
+      procs.append(proc)
+      while (childCount() > num_workers):
+        time.sleep(0.25)
+        [pp.communicate() for pp in procs] # this will get the exit code
+        procs = []
+      else:
+        if (i == len(lstcmd)-1):
+          try:
+            outs, errs = proc.communicate()
+            if proc.returncode == 0:
+              print("stdout = {}; stderr = {}".format(str(outs),str(errs)))
+            else:
+              exit("ERROR: subprocess {} failed".format(str(lstcmd[i])))
+          except:
+            break
+    else:
+      return_code = proc.communicate()
+      if return_code != 0:
+        exit("Failed to run {}".format(str(p)))
+  #set a delay to avoid delay in writing process
+  time.sleep(1)
+  print("done submitting")
+  del(lstcmd)
+{%- endif %}
+
+{%- if "enso" in subset %}
+  #####################################################################
+  # call enso_driver.py to process diagnostics
+  #####################################################################
+  print("calculate enso metrics")
+  groups = list({{ groups }})
+  lstcmd = []
+  for metricsCollection in groups:
+    cmd = (" ".join(['enso_driver.py',
+                      '-p', "parameterfile.py",
+                      '--metricsCollection',metricsCollection]))
+    lstcmd.append(cmd); del(cmd)
+  #finally process the data in parallel
+  print("Number of jobs starting is ", str(len(lstcmd)))
+  procs = []
+  for i,p in enumerate(lstcmd):
+    print('running %s' % (str(p)))
+    proc = Popen(p, stdout=PIPE, shell=True)
+    procs.append(proc)
+    while (childCount() > {{num_workers}}):
+      time.sleep(0.25)
+      [pp.communicate() for pp in procs] # this will get the exit code
+      procs = []
+    else:
+      if (i == len(lstcmd)-1):
+        try:
+          outs, errs = proc.communicate()
+          if proc.returncode == 0:
+            print("stdout = {}; stderr = {}".format(str(outs),str(errs)))
+          else:
+            exit("ERROR: subprocess {} failed".format(str(lstcmd[i])))
+        except:
+          break
+  #set a delay to avoid delay in writing process
+  time.sleep(1)
+  print("done submitting")
+  del(lstcmd,procs)
+  #post process and generate figures
+  #print("mip, exp, MC, case_id:", '{{ mip}}', '{{ exp }}', metricsCollection, '${case_id}')
+  #merge_jsons('{{ mip }}', '{{ exp}}', '${case_id}', metricsCollection, '${results_dir}')
+{%- endif %}
+
+if __name__ == "__main__":
+  main()
+
+EOF
+
+################################
+# Run diagnostics
+#command="srun -n 1 python -u pcmdi.py"
+command="python -u pcmdi.py"
+# Run diagnostics
+time ${command}
+if [ $? != 0 ]; then
+  cd {{ scriptDir }}
+  echo 'ERROR (10)' > {{ prefix }}.status
+  exit 9
+fi
+
+# Copy output to web server
+echo
+echo ===== COPY FILES TO WEB SERVER =====
+echo
+
+# Create top-level directory
+web_dir=${www}/${case}/pcmdi_diags #/{{ sub }}
```

Comment:
We had discussed, for production rather than prototyping, that we'd only put images on the `www` path, and keep everything else on the `output` path. This looks like it's copying everything over to the web server, is that right?

Reply:
@forsyth2 : Yes, I have currently copied all output from PCMDI to the www path as I need to talk with you and Chengzhu to figure out the right sets of data that should be put in the www path.  Currently, I followed the E3SM_diags, as some of the diagnostic data from E3SM_diags will also be copied to www directory.